### PR TITLE
Add pod name/namespace filter in volume list

### PR DESCRIPTION
**Pull request by balamurugana**
Signed-off-by: Bala.FA <bala.gluster@gmail.com>

### CODE REVIEW

File: cmd/kubectl-direct_csi/volumes_list.go [hunk 1]
```diff
@@ -65,6 +70,8 @@ func init() {
 	listVolumesCmd.PersistentFlags().StringSliceVarP(&nodes, "nodes", "n", nodes, "glob prefix match for node names")
 	listVolumesCmd.PersistentFlags().StringSliceVarP(&volumeStatus, "status", "s", volumeStatus, "filters based on volume status. The possible values are [staged,published]")
 	listVolumesCmd.PersistentFlags().StringSliceVarP(&accessTiers, "access-tier", "", accessTiers, "filter based on access-tier")
+	listVolumesCmd.PersistentFlags().StringSliceVarP(&podNames, "pod-name", "", podNames, "glob prefix match for pod names")
```
Comment (Praveenrajmani) [inline cmd/kubectl-direct_csi/volumes_list.go:73]: Can we add some examples in the helpers above?

**Review by Praveenrajmani — DISMISSED** [review]
Tested and LGTM

**Comment by Praveenrajmani** [comment]
Also, Can we show `pod-name`, `pod-namespace` and `status` in the tabular output for `volumes ls`? @balamurugana @wlan0

File: cmd/kubectl-direct_csi/volumes_list.go (hunk 1 above)
Comment (balamurugana) [reply cmd/kubectl-direct_csi/volumes_list.go:79]: Done

**Comment by balamurugana** [comment]
> Also, Can we show `pod-name`, `pod-namespace` and `status` in the tabular output for `volumes ls`? @balamurugana @wlan0

Done

File: cmd/kubectl-direct_csi/volumes_list.go [hunk 2]
```diff
@@ -51,6 +56,12 @@ $ kubectl direct-csi volumes ls --nodes=directcsi-1
 
 # Combine multiple filters using csv
 $ kubectl direct-csi vol ls --nodes=directcsi-1,directcsi-2 --status=staged --drives=/dev/nvme0n1
+
+# List all volumes of pod name
+$ kubectl direct-csi volumes ls --pod-name=my-minio*
```
Comment (Praveenrajmani) [inline cmd/kubectl-direct_csi/volumes_list.go:61]: The `pod.name` and `pod.namespace` labels will be available only on "published" volumes. Can we explain that in the example here. Like, 

```
# List volumes published for the provided pods
$ kubectl direct-csi volumes ls --pod-name=my-minio* --status=published

# List volumes published in the provided namespaces
kubectl direct-csi volumes ls --pod-namespace=my-minio-ns* --status=published
```

File: cmd/kubectl-direct_csi/volumes_list.go [hunk 3]
```diff
@@ -150,6 +165,8 @@ func listVolumes(ctx context.Context, args []string) error {
 			emptyOrBytes(v.Status.TotalCapacity),  //CAPACITY
 			v.Status.NodeName,                     //SERVER
 			driveName(drivePaths[v.Status.Drive]), //DRIVE
+			v.ObjectMeta.Labels[directcsi.Group+"/pod.name"],
```
Comment (Praveenrajmani) [inline cmd/kubectl-direct_csi/volumes_list.go:168]: Incase if the value is empty - "" , Can we show an hiphen instead? Like we do for the drives listing (showing total capacity and allocated capacity) 

```
cleanLabel := func(val string) string {
	if val == "" {
		return "-"
	}
	return val
}
```

**Review by Praveenrajmani — COMMENTED** [review]
Few minor comments/suggestions

File: cmd/kubectl-direct_csi/volumes_list.go [hunk 4]
```diff
@@ -150,6 +165,8 @@ func listVolumes(ctx context.Context, args []string) error {
...
+			v.ObjectMeta.Labels[directcsi.Group+"/pod.namespace"],
```
Comment (Praveenrajmani) [inline cmd/kubectl-direct_csi/volumes_list.go:169]: Will add the `status` display in another PR :+1:

File: cmd/kubectl-direct_csi/volumes_list.go (hunk 2 above)
Comment (balamurugana) [reply cmd/kubectl-direct_csi/volumes_list.go]: Done

File: cmd/kubectl-direct_csi/volumes_list.go (hunk 3 above)
Comment (balamurugana) [reply cmd/kubectl-direct_csi/volumes_list.go]: Done

**Comment by wlan0** [comment]
LGTM

**Review by wlan0 — APPROVED** [review]

**Pull request merged**